backend/src/routes/chat/chat.controller.ts

```diff
@@ -36,13 +37,38 @@ export class ChatController {
     return this.chatService.getPublicChannels();
   }
 
+  @Post("is_admin/:name")
+  async isadmin(
+    @Req() req: Request,
+    @Param("name") name: string,
+    @Body("chan") chan: string
+  ) {
+    return this.chatService.is_admin(name, chan);
+  }
+  @Post("is_muted/:name")
+  async ismuted(
+    @Req() req: Request,
+    @Param("name") name: string,
+    @Body("chan") chan: string
+  ) {
+    return this.chatService.ismuted(name, chan);
+  }
+  @Post("is_banned/:name")
+  async isbanned(
+    @Req() req: Request,
+    @Param("name") name: string,
+    @Body("chan") chan: string
+  ) {
+    return this.chatService.isBanned(name, chan);
+  }
   @Post("join/:name")
   async joinChannel(
     @Req() req: Request,
     @Param("name") name: string,
     @MessageBody("password") password: string,
     @MessageBody("public") pub: boolean
   ) {
+    if (!req || !name) return;
```

Comment:
No need to check req, it should never be null. We also need to throw an HTTPException instead of returning there and in all the following functions
```suggestion
    if (!name) throw new HTTPException("Invalid name", HTTPStatus.BAD_REQUEST);
```